sentry_sdk/integrations/qdrant/qdrant.py

```diff
@@ -0,0 +1,523 @@
+        return
+
+    parsed_url = urlparse(httpx_url)
+    httpx_path = parsed_url.path
+
+    # just to be *really* sure that we don't accidentally delete an unrelated span
+    httpx_operation_id = _qdrant_trie.match(httpx_path, httpx_method)
+
+    if httpx_operation_id == operation_id:
+        span_recorder.spans.pop(next_span_index)
+
+
+@contextmanager
+def _prepare_span(request, mute_http_child_span):
+    # type: (Request, bool) -> Span
+    operation_id = _qdrant_trie.match(request.url.path, request.method)
+    payload = json.loads(request.content) if request.content else {}
+    payload["operation_id"] = operation_id
+
+    name = json.dumps(_strip_dict(payload, _DISALLOWED_REST_FIELDS), default=str)
+
+    span = sentry_sdk.start_span(
+        op=OP.DB_QDRANT_REST,
+        name=name,
+        origin=_IDENTIFIER,
+    )
+    span.set_data("db.api", "REST")
+    span.set_data(SPANDATA.DB_SYSTEM, "qdrant")
+    span.set_data(SPANDATA.DB_OPERATION, operation_id)
+
+    collection_name = getattr(request, "collection_name", None)
+    if collection_name:
+        span.set_data("db.qdrant.collection", collection_name)
+
+    yield span
+
+    # the HttpxIntegration will capture all REST calls, leading to almost duplicate spans but with less information.
+    # we mute the created span by httpx by default, but you can disable this in the Qdrant integration if you want.
+    if mute_http_child_span:
+        _remove_httpx_span(span, operation_id)
+
+    span.finish()
+
+
+def _get_integration():
+    # type: () -> (Optional[QdrantIntegration])
+    from sentry_sdk.integrations.qdrant import QdrantIntegration
+
+    return sentry_sdk.get_client().get_integration(QdrantIntegration)
+
+
+def _sync_api_client_send_inner(f):
+    # type: (Callable[P, Response]) -> Callable[P, Response]
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        # type: (P.args, P.kwargs) -> Response
+        integration = _get_integration()  # type: Optional[QdrantIntegration]
+        if integration is None or not (len(args) >= 2 and isinstance(args[1], Request)):
+            return f(*args, **kwargs)
+
+        request = args[1]  # type: Request
+        with _prepare_span(request, integration.mute_children_http_spans):
+            return f(*args, **kwargs)
+
+    return wrapper
+
+
+def _async_api_client_send_inner(f):
+    # type: (Callable[P, Awaitable[Response]]) -> Callable[P, Awaitable[Response]]
+    @wraps(f)
+    async def wrapper(*args, **kwargs):
+        # type: (P.args, P.kwargs) -> Response
+        integration = _get_integration()  # type: Optional[QdrantIntegration]
+        if integration is None or not (len(args) >= 2 and isinstance(args[1], Request)):
+            return f(*args, **kwargs)
+
+        request = args[1]  # type: Request
+        with _prepare_span(request, integration.mute_children_http_spans):
+            return await f(*args, **kwargs)
+
+    return wrapper
+
+
+# taken from grpc integration
```

Comment:
Why not just use the grpc integration (or use this code from the grpc integration) directly?